lib/matplotlib/axes/_axes.py

```diff
@@ -4605,6 +4552,10 @@ def fill(self, *args, **kwargs):
                 ax.fill(x, y, x2, y2)            # two polygons
                 ax.fill(x, y, "b", x2, y2, "r")  # a blue and a red polygon
 
+        data : indexable object, optional
+            An object with labelled data. If given, provide the label names to
+            plot in *x* and *y*.
```

Comment:
This could use an 'e.g.' `ax.fill(x='time', y='signal', data=thedata)`

Reply:
done